### ISSUE TITLE
Coreference model trained with Allennlp v0.5.1 gets small scores

### ISSUE THREAD

**Issue by ljinstat** [allenai/allennlp#1545]
Hello!

I trained the coreference model with Allennlp 0.5.1 and get F1 score around 0.09 which is far smaller than the score in the paper (about 0.68).

For Allennlp 0.5.1, Python 3.6, Pytorch 0.4.0 are required. I ran it on GPU Tesla P100. It works well when the model is trained with v0.4.3. 

Could you please tell me your performance of the coreference model trained with v0.5.1? I don't know whether it exists a problem somewhere.
Thanks!

**Comment by schmmd** [comment]
@DeNeutoy would you be able to kick off a Beaker job to retrain the coreference model so we can verify this issue?  I think it will be easiest for you, since the datasets are not publicly available.

**Comment by apeterswu** [comment]
@ljinstat hi, I tried to run coref model with 0.4.3 as you mentioned, but the performance only got 61.8 on test set: 
  "test_coref_precision": 0.6744485009227493,
  "test_coref_recall": 0.5709880690236084,
  "test_coref_f1": 0.6180333449383129,
  "test_mention_recall": 0.9197024893746205

Seems far away from 67.2, do you have any suggestion? Thanks.

**Comment by ljinstat** [comment]
Hi @apeterswu, my test result with 0.4.3 is 61.4, even lower than yours. Generally, we cannot reproduce results as good as that in papers. Besides, I think the maximum number of antecedents really matters to the results. Due to memory issues, maximum antecedents are 100, while in paper 250 antecedents are considered. By the way, the result in the paper are produced by two GPUs.

Modifications that you can improve the result are from the answer of another issue #1337: 
> 1. Using 840B.300d glove vectors instead of the 6B ones, 2. Using the bidirectional LSTM in allennlp which has variational dropout and 3 adding speaker features. The first two are trivially easy (you can change them in the config) and adding speaker features only requires a little bit of effort.

Hope it helps.

**Comment by apeterswu** [comment]
@ljinstat thanks for your response. I will make a try.

**Comment by ljinstat** [comment]
Hi @apeterswu, yes I have implemented the EMLo. The essential thing is to calculate correctly the dimension of each layer. I will upload the configuration file and the explanation later on my github, so that anyone who needs it can make a reference. Otherwise, you can read the paper and the tutorial of Allennlp about EMLo.

**Comment by apeterswu** [comment]
@ljinstat Thanks, yes, what I am not sure is the dimension. Ok, I will make a try first, also look forward to your github. :)

**Comment by matt-gardner** [comment]
We've done some initial exploration here, and our guess at this point is that the model is still behaving the same, but the metric computation ([here](https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/conll_coref_scores.py)) silently broke when we upgraded to pytorch 0.4, because that class doesn't have any tests.  We think this because a before-and-after on tensorboard looks like the model parameters, loss, and mention recall metrics are all indistinguishable between the two versions, but the metrics computed by `ConllCorefScores` are all way off (which shouldn't be possible if the loss is the same).

**Comment by VictorSanh** [comment]
Awesome, thank you very much for the fix ! (And more generally thank you for this amazing library !)

**Comment by ljinstat** [comment]
Thank you for the fix!
